src/main/java/javacake/storage/Profile.java

```diff
@@ -0,0 +1,158 @@
+package javacake.storage;
+
+import javacake.Duke;
+import javacake.exceptions.DukeException;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.logging.Level;
+
+public class Profile {
+    private static String filepath = "data/save/savefile.txt";
+    private String username;
+    private ArrayList<Integer> topicsDone = new ArrayList<>();
+
+    public Profile() throws DukeException {
+        this("data/");
+    }
+
+    /**
+     * Constructor for profile.
+     * @param filename String of filepath
+     * @throws DukeException when unable to create profile
+     */
+    public Profile(String filename) throws DukeException {
+        try {
+            File file = new File("data/save/savefile.txt");
+            Duke.logger.log(Level.INFO,"Filepath: " + filepath);
+            try {
+                if (!file.getParentFile().getParentFile().exists()) {
+                    file.getParentFile().getParentFile().mkdir();
+                    file.getParentFile().mkdir();
+                    file.createNewFile();
+                    initialiseUser();
+                    System.out.println("A" + file.getParentFile().getParentFile().getPath());
+                } else if (!file.getParentFile().exists()) {
+                    file.getParentFile().mkdir();
+                    file.createNewFile();
+                    initialiseUser();
+                    System.out.println("B" + file.getParentFile().getPath());
+                } else if (!file.exists()) {
+                    file.createNewFile();
+                    initialiseUser();
+                    System.out.println("C" + file.getPath());
+                } else {
+                    Duke.logger.log(Level.INFO, filepath + " is found!");
+                }

```

Comment:
Good effort in keeping this neat and nicely formatted. However, having some comments here would be nice to help the developer understand your code.